gemrb/GUIScripts/pst/GUIREC.py

```diff
@@ -819,6 +819,9 @@ def AcceptLevelUp():
 	#increase weapon proficiency if needed
 	if WeapProfType!=-1:
 		GemRB.SetPlayerStat (pc, IE_PROFICIENCYBASTARDSWORD+WeapProfType, CurrWeapProf + WeapProfGained )
+	else:
+		freeSlots = GemRB.GetPlayerStat(pc, IE_FREESLOTS)
+		GemRB.SetPlayerStat (pc, IE_FREESLOTS, freeSlots + WeapProfGained )
```

Comment:
WeapProfGained isn't decreased any where, so shouldn't this be a subtraction, capped at 0?

Reply:
WeapProfGained should just be the total new prof points gained from the ```HasGainedWeapProf``` loop. You don't get to actually assign them immediately on level up. I thought to just treat it like any other weapon proficiency, because you can potentially stack them up if you can't find an NPC that can train you. But it's quite possible I have this sussed all wrong...